xapian-letor/tests/api_letor.cc

```diff
@@ -182,6 +182,48 @@ DEFINE_TESTCASE(svm_ranker, generated)
     return true;
 }
 
+DEFINE_TESTCASE(listmle_ranker, generated)
+{
+    Xapian::ListMLERanker ranker;
+    TEST_EXCEPTION(Xapian::FileNotFoundError, ranker.train_model(""));
+    string db_path = get_database_path("apitest_listmle_ranker",
+				       db_index_two_documents);
+    Xapian::Enquire enquire((Xapian::Database(db_path)));
+    enquire.set_query(Xapian::Query("lions"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    string data_directory = test_driver::get_srcdir() + "/testdata/";
+    string query = data_directory + "query.txt";
+    string qrel = data_directory + "qrel.txt";
+    Xapian::prepare_training_file(db_path, query, qrel, 10,
+				  "listmle_ranker_training.txt");
```

Comment:
The same comment would apply about the `filename` argument to `::prepare_training_file()` as to the other path arguments: they should be relative to `test_driver::get_srcdir()`. This change should be made in this PR. (Even if we're going to rework this in a future change.)